public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_traces/app_insights_traces_datasource.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable*/
+// TODO: remove lint disable rule
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import {
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceInstanceSettings,
+  MutableDataFrame,
+  Field,
+  TraceKeyValuePair,
+  Vector,
+} from '@grafana/data';
+
+import AzureLogAnalyticsDatasource from '../azure_log_analytics/azure_log_analytics_datasource';
+import { AzureDataSourceJsonData, AzureMonitorQuery } from '../types';
+
+export default class AzureTracesDatasource extends AzureLogAnalyticsDatasource {
+  operationId: string;
+  constructor(instanceSettings: DataSourceInstanceSettings<AzureDataSourceJsonData>) {
+    super(instanceSettings);
+    this.operationId = '';
+  }
+
+  query(request: DataQueryRequest<AzureMonitorQuery>): Observable<DataQueryResponse> {
+    this.operationId = request.targets[0].azureLogAnalytics?.operationId ?? '';
+    // only take the first query
+    const target = this.buildTraceQuery(request.targets[0]);
+
+    const req = { ...request, targets: [target] };
+    return super.query(req).pipe(
+      map((res: DataQueryResponse) => {
+        const newDf = this.convertResponseToTrace(res);
+        return newDf;
+      })
+    );
+  }
+  buildTraceQuery(target: AzureMonitorQuery): AzureMonitorQuery {
+    if (!target.azureLogAnalytics || !this.operationId || this.operationId === '') {
+      return { ...target };
+    }
+    const queryString = `set truncationmaxrecords=10000;
+    set truncationmaxsize=67108864;
+    union isfuzzy=true customEvents,pageViews,requests,dependencies,exceptions,availabilityResults
+    | where timestamp > $__timeFrom and timestamp < $__timeTo
+    | where (operation_Id != '' and operation_Id == '${this.operationId}') or (customDimensions.ai_legacyRootId != '' and customDimensions.ai_legacyRootId == '${this.operationId}')
```

Comment:
the logic here (operation_Id != '' and operation_Id == '${this.operationId}') doesn't sound that good. pls add a to-do to improve

Reply:
Will do, right now I added it so that no data is returned when there is no operation Id provided. Mainly because when a template variable is used, the operation ID is technically not empty even if the variable is.